subprojects/plugin-use/src/main/java/org/gradle/plugin/management/internal/autoapply/DefaultAutoAppliedPluginHandler.java

```diff
@@ -41,27 +41,26 @@ public DefaultAutoAppliedPluginHandler(AutoAppliedPluginRegistry registry) {
         this.registry = registry;
     }
 
-    @Override
-    public PluginRequests mergeWithAutoAppliedPlugins(PluginRequests initialRequests, Object pluginTarget) {
+    public PluginRequests getAutoAppliedPlugins(Object pluginTarget) {
         if (pluginTarget instanceof Project) {
             Project project = (Project) pluginTarget;
 
             PluginRequests autoAppliedPlugins = registry.getAutoAppliedPlugins(project);
             if (autoAppliedPlugins.isEmpty()) {
-                return initialRequests;
+                return PluginRequests.EMPTY;
             }
-            return mergePluginRequests(autoAppliedPlugins, initialRequests, project.getPlugins(), project.getBuildscript());
+            return autoAppliedPlugins;
         } else if (pluginTarget instanceof Settings) {
             Settings settings = (Settings) pluginTarget;
 
             PluginRequests autoAppliedPlugins = registry.getAutoAppliedPlugins(settings);
             if (autoAppliedPlugins.isEmpty()) {
-                return initialRequests;
+                return PluginRequests.EMPTY;
             }
-            return mergePluginRequests(autoAppliedPlugins, initialRequests, settings.getPlugins(), settings.getBuildscript());
```

Comment:
I should probably still filter out the auto-applied plugins at this point, similar to the previous behavior. I will add it.

Reply:
Done